src/dispatch/plugins/dispatch_slack/case/messages.py

```diff
@@ -298,6 +305,116 @@ def create_signal_messages(case_id: int, channel_id: str, db_session: Session) -
     return Message(blocks=signal_metadata_blocks).build()["blocks"]
 
 
+def create_genai_signal_summary(
+    case: Case,
+    channel_id: str,
+    db_session: Session,
+    client: WebClient,
+) -> list[Block]:
+    """Creates enhanced signal messages with historical context."""
+    signal_metadata_blocks: list[Block] = []
+
+    signal_instances_query = (
+        db_session.query(SignalInstance, Signal)
+        .join(Signal)
+        .with_entities(SignalInstance.id, Signal)
+        .filter(SignalInstance.case_id == case.id)
+        .order_by(SignalInstance.created_at)
+    )
+
+    (first_instance_id, first_instance_signal) = signal_instances_query.first()
+
+    # Fetch up to 10 recent related cases
```

Comment:
I'd suggest we move fetching the cases and related historical context outside of this function, so it can be reused for other case related use cases.